src/query/plan/read_write_type_checker.hpp

```diff
@@ -87,6 +87,13 @@ struct ReadWriteTypeChecker : public virtual HierarchicalLogicalOperatorVisitor
   bool PreVisit(CallProcedure &) override;
   bool PreVisit(Foreach &) override;
 
+  bool PreVisit(Apply &) override;
+  bool PreVisit(IndexedJoin &) override;
+  bool PreVisit(HashJoin &) override;
+  bool PreVisit(RollUpApply &) override;
```

Comment:
Why did these get introduced?

Reply:
it was a random bug that I encountered when trying to do a write query on a replica. Apparently since these didn't get written, I was able to perform a write query on the replica for the PeriodicSubquery. This is the leftover from other PR's since, for example CALL didn't get inspected if inside there was some write operation.